File: app/src/test/java/bj/rxjavaexperimentation/schedulerprovider/TestSchedulerProvider.java
package bj.rxjavaexperimentation.schedulerprovider;

import io.reactivex.Scheduler;
import io.reactivex.schedulers.TestScheduler;

/**
 * Created by Josh Laird on 11/04/2017.
 */
public class TestSchedulerProvider extends MySchedulerProvider
{
    private final TestScheduler testScheduler;

    public TestSchedulerProvider(final TestScheduler testScheduler)
    {
        this.testScheduler = testScheduler;
    }

    @Override
    public Scheduler ui()
    {
        return testScheduler;
    }

    @Override
    public Scheduler computation()
    {
        return testScheduler;
    }

    @Override
    public Scheduler trampoline()
    {
        return testScheduler;
    }

    @Override
    public Scheduler newThread()
    {
        return testScheduler;
    }

    @Override
    public Scheduler io()
    {
        return testScheduler;
    }
}

File: app/src/test/java/bj/discogsbrowser/utils/schedulerprovider/TestSchedulerProvider.kt
package bj.discogsbrowser.utils.schedulerprovider

import io.reactivex.Scheduler
import io.reactivex.schedulers.TestScheduler

/**
 * Created by Josh Laird on 22/05/2017.
 */
class TestSchedulerProvider(val testScheduler: TestScheduler) : MySchedulerProvider() {
    override fun ui(): Scheduler {
        return testScheduler
    }

    override fun computation(): Scheduler {
        return testScheduler
    }

    override fun trampoline(): Scheduler {
        return testScheduler
    }

    override fun newThread(): Scheduler {
        return testScheduler
    }

    override fun io(): Scheduler {
        return testScheduler
    }
}